### PR TITLE
refactor: remove graphql and add prisma and express

### PR DESCRIPTION
**Pull request by talo242**
### What this PR does

This PR does the following:
- It removes GraphQL and Apollo from the project
- Adds Postgres, Prisma, Express, and Docker to the project
- Adds a basic endpoint `policies` with one query param `search`
- Updates the README.md file

### Why is this needed?
We want to make the coding challenge easier to set up.

### Next steps:
- [ ] Update Readme files on backend and frontend folders
- [x] Fix migration and seed commands inside of docker
- [ ] Add basic styles to Frontend

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  lastName      String      @db.VarChar(255)
+  dateOfBirth   DateTime
+  policies      Policy[]
+}
+
+model Policy {
+  id            String        @id @default(uuid()) @db.Uuid
+  customer      Customer      @relation(fields: [customerId], references: [id])
+  customerId    String        @db.Uuid
+  provider      String
```
Comment (killthekitten) [inline backend/prisma/schema.prisma:38]: This is not critical, but provider could be an enum too, exactly the way `insuranceType` is set up.

**Review by killthekitten — APPROVED** [review]
Looks like a really good start for the backend! Just ran it – works like a charm.

Did I get it correctly, that for the FE part there's going to be a separate PR? Right now it looks like the backend part has a bit of a head start compared to the blank `App.tsx` on FE.  

TIL: running `docker compose up` in a subdir (`backend`) picks up the `docker-compose.yaml` from a parent dir 👍

**Comment by killthekitten** [comment]
@talo242 actually, taking down docker compose in VSCode is broken. It attempts to do so from the root dir, while `.env`  lives under /backend. Perhaps you could move `.env` next to `docker-compose.yaml`? 

It's obviously a trade-off, but it might remove some confusion. Anyone who would want to run the apps independently without docker should be instructed to copy .env.example from the root dir to the respective frontend/backend folder.

**Review by vincentaudoire — APPROVED** [review]

**Review by mbaguiar — APPROVED** [review]
I tried to run everything, and it worked as expected ✨

I just feel the boilerplate we provider for the backend vs. the one for the frontend is not really balanced. However, I don't really know what we could add without it being too much

**Comment by talo242** [comment]
> Did I get it correctly, that for the FE part there's going to be a separate PR? Right now it looks like the backend part has a bit of a head start compared to the blank App.tsx on FE.

Yes, we decided to add some basic styles to that page, so it's a little bit more balanced.
The task should focus on implementing the search, so I think we can add some styles and components taken from Tailwind (like the menu and the table).

We discussed that maybe is not a good idea to add API calls because that can bias the implementation.

> @talo242 actually, taking down docker compose in VSCode is broken. It attempts to do so from the root dir, while .env lives under /backend. Perhaps you could move .env next to docker-compose.yaml?

Thanks for the heads up, I'll take a look at it!

File: Readme.md [hunk 2]
```diff
@@ -1,64 +1,117 @@
...
+You can spend as little or as much time as you like on this project. We've generated some boilerplate for you to get started.
 
-- How you write idiomatic code.
-- How you deal with UI/UX.
-- How you navigate the requirements.
-- How you test & write testable code.
-- How you package the code.
-- How you deal with the security concerns.
-- How you communicate with your fellow programmers.
+1. Start by reading the [User story](#User-story) and the [Acceptance criteria](#Acceptance-criteria) have a clear idea of the requirements.
```
[the remark this answers is not in the document]
Comment (talo242) [reply Readme.md]: Good eye, updated 👍🏼

**Comment by talo242** [comment]
> @talo242 actually, taking down docker compose in VSCode is broken. It attempts to do so from the root dir, while .env lives under /backend. Perhaps you could move .env next to docker-compose.yaml?

It seems like Prisma doesn't like when you move the `.env` file outside of the root directory. So I decided to add the `env_file` parameter to the docker-compose file. This should prevent the IDEs to get confused 😄

**Pull request merged**